functional/FunctionalTest.java

```diff
@@ -1074,11 +1074,17 @@ public static void copyObject_test7() throws Exception {
 
     copyConditions.setUnmodified(dateRepresentation);
 
+    ErrorResponseException expectedException = null;
+
     try {
       client.copyObject(bucketName, fileName, secondBucketName, copyConditions);
     } catch (ErrorResponseException e) {
       // File should not be copied as object was modified after date set in copyConditions.
-      ignore();
+      expectedException = e;
+    }
+
+    if (expectedException == null || !expectedException.errorResponse().code().equals("PreconditionFailed")) {
```

Comment:
Its not java style. Can you tell me a case where its useful?
Catching and raising its own exception is unwanted thing in functional test.  The scope of the test is up to what expected. If anything happens beyond that is an error.